src/backend/columnar/columnar_tableam.c

```diff
@@ -740,6 +740,19 @@ columnar_tuple_insert(Relation relation, TupleTableSlot *slot, CommandId cid,
 	ColumnarWriteState *writeState = columnar_init_write_state(relation,
 															   RelationGetDescr(relation),
 															   GetCurrentSubTransactionId());
```

Comment:
What about adding another param to `columnar_init_write_state()` such that it becomes:
```diff
ColumnarWriteState *
columnar_init_write_state(Relation relation, TupleDesc tupdesc,
+++						  Oid tupSlotRelationId,
						  SubTransactionId currentSubXid)
```

Given that `columnar_init_write_state()` initializes only at once, this should let it decide correct `ColumnarOptions` at the very first call.

Question: For multi-insert's (columnar_multi_insert), there would be an array of `TupleTableSlot`s. In that case, would it be okay to pass `slot->tts_tableOid` for the first tuple slot ?